ykshim_client/src/test_api.rs

```diff
@@ -0,0 +1,138 @@
+//! The testing API client to ykshim.
+
+use crate::prod_api::{Local, RawSirTrace, SirTrace, TyIndex};
+use libc::size_t;
+use std::collections::HashMap;
+use std::ffi::{c_void, CString};
+use std::fmt;
+use std::os::raw::c_char;
+
+// Opaque pointers.
+type RawTirTrace = c_void;
+type RawTraceCompiler = c_void;
+
+// Keep these types in-sync with the internal workspace.
+#[derive(Debug, Clone, Copy)]
+#[repr(C)]
+pub struct CguHash(u64);
+// FIMXE make this a repr(C) struct so as to simplify __ykshimtest_body_ret_ty.
+pub type TypeId = (CguHash, TyIndex);
+
+extern "C" {
+    fn __ykshimtest_sirtrace_len(sir_trace: *mut RawSirTrace) -> size_t;
+    fn __ykshimtest_tirtrace_new(sir_trace: *mut RawSirTrace) -> *mut RawTirTrace;
+    fn __ykshimtest_tracecompiler_drop(comp: *mut RawTraceCompiler);
+    fn __ykshimtest_tirtrace_len(tir_trace: *mut RawTirTrace) -> size_t;
+    fn __ykshimtest_tirtrace_display(tir_trace: *mut RawTirTrace) -> *mut c_char;
+    fn __ykshimtest_body_ret_ty(sym: *mut c_char, ret_cgu: *mut CguHash, ret_idx: *mut TyIndex);
+    fn __ykshimtest_tracecompiler_default() -> *mut RawTraceCompiler;
+    fn __ykshimtest_tracecompiler_insert_decl(
+        tc: *mut RawTraceCompiler,
+        local: Local,
+        local_ty_cgu: CguHash,
+        local_ty_index: TyIndex,
+        referenced: bool,
+    );
+    fn __ykshimtest_tracecompiler_local_to_location_str(
+        tc: *mut RawTraceCompiler,
+        local: Local,
+    ) -> *mut c_char;
+    fn __ykshimtest_tracecompiler_local_dead(tc: *mut RawTraceCompiler, local: Local);
+    fn __ykshimtest_tracecompiler_find_sym(sym: *mut c_char) -> *mut c_void;
+    fn __yktest_interpret_body(body_name: *mut c_char, icx: *mut u8);
+    fn __yktest_reg_pool_size() -> usize;
```

Comment:
Why did you mix `__ykshimtest` and `__yktest`?

Reply:
That's a mistake on my part. It's supposed to be `__ykshimtest` across the board. Will fix.

Reply:
Fixed in 124cd35.